### PR TITLE
Make the Brightness estimator public

### PR DESCRIPTION
**Pull request by Hixie**
cc @HansMuller

### CODE REVIEW

File: packages/flutter/lib/src/material/theme_data.dart [hunk 1]
```diff
@@ -467,6 +441,37 @@ class ThemeData {
...
+  }
+
+  /// Determines whether the given [Color] is [Brightness.light] or
+  /// [Brightness.dark].
+  ///
+  /// This compares the luminosity of the given color to a threshold value that
+  /// matches the material design specification.
+  static Brightness estimateBrightnessForColor(Color color) {
+    // See <https://www.w3.org/TR/WCAG20/#relativeluminancedef>
+    final double R = _linearizeColorComponent(color.red / 0xFF);
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/theme_data.dart]: I'm surprised that it's necessary to mask of the lower 8 bits. I thought Color.foo already masked off 8 bits

Comment (Hixie) [inline packages/flutter/lib/src/material/theme_data.dart]: That line doesn't mask off the low 8 bits, it converts the integer representation (0 - 255) to a real interpretation (0.0 - 1.0).

Comment (HansMuller) [inline packages/flutter/lib/src/material/theme_data.dart]: Oops, that was a '/' not an &.

**Comment by HansMuller** [comment]
examples/flutter_gallery/lib/demo/colors_demo.dart does a related computation that we might want to simplify at some point: given some (background) color, would black or white provide more contrast.

**Comment by HansMuller** [comment]
LGTM

**Pull request merged**